monitor/data_monitoring.py

```diff
@@ -77,25 +88,49 @@ def _query_alerts(self, days_back: int) -> list:
                 if test_result_object:
                     test_result_alerts.append(test_result_object)
                 else:
+                    failed_to_parse_alert_dicts.append(test_result_alert_dict)
                     self.success = False
 
-        return test_result_alerts
+        return AlertsQueryResult(test_result_alerts, failed_to_parse_alert_dicts)
 
     def _send_to_slack(self, test_result_alerts: List[TestResult]) -> None:
         sent_alerts = []
         alerts_with_progress_bar = alive_it(test_result_alerts, title="Sending alerts")
         for alert in alerts_with_progress_bar:
-            alert_slack_message: SlackMessageSchema = alert.generate_slack_message(is_slack_workflow=self.config.is_slack_workflow)
+            alert_slack_message = alert.generate_slack_message(is_slack_workflow=self.config.is_slack_workflow)
             sent_successfully = self.slack_client.send_message(
                 channel_name=self.slack_channel_name,
                 message=alert_slack_message
             )
             if sent_successfully:
                 sent_alerts.append(alert.id)
             else:
-                logger.error(f"Could not sent the alert - {alert.id}. Full alert: {json.dumps(dict(alert_slack_message))}")
+                logger.error(
+                    f"Could not sent the alert - {alert.id}. Full alert: {json.dumps(dict(alert_slack_message))}")
                 self.success = False
-        
+
+        sent_alert_count = len(sent_alerts)
+        self.execution_properties['sent_alert_count'] = sent_alert_count
+        if sent_alert_count > 0:
+            self._update_sent_alerts(sent_alerts)
+
+    def _send_raw_dicts_to_slack(self, alert_dicts: List[dict]) -> None:
```

Comment:
So this method is really similar to the `_send_to_slack` method.
the only thing that actually different is the way we create the message itself.
I think it will be better to merge those methods into one and create a method for creating the message.
Plus in your case, the error alerts won't be calculated inside of the `alerts_with_progress_bar` which we do want to count :)